### DISCUSSION TITLE
AI based troubleshooting extensions using long term DB metrics

### DISCUSSION THREAD

**Discussion by erdem2000** [netdata/netdata#12121] [Ideas]
Hi folks ! 
Actually, this idea is not directly related with an potential agent feature and there is already "ml" support for real-time analyzing purposes. But we can consider it as an collaborative tool for troubleshooting.

To make Netdata more powerful (in the near future), it could be useful to make inferences using AI algorithms such as LSTM to recognize troubleshoot patterns from long term metrics which are stored in the exported databases. These AI extensions can be used by the registered users.

Analyzed outputs could be used in several fields such as predicting future troubleshoots and their origins, feedback to the real-time ml algos in the agent and statistical information.

![idea1](https://user-images.githubusercontent.com/51407/153601978-c8e1af50-d9a2-4976-a1cc-3ee20b6118f9.png)

**Comment by shyamvalsan** [comment]
Hi @erdem2000 thanks for sharing! This is an area we are working on. One of the key aspects as you mention is the capability for "longer term" metric storage.
